nmostesting/GenericTest.py

```diff
@@ -114,6 +116,9 @@ def __init__(self, apis, omit_paths=None, disable_auto=False):
 
         self.result.append(test.NA(""))
 
+        self.primary_auth = None if not auths else auths[0]
+        self.secondary_auth = None if not auths else auths[1]
+
```

Comment:
Idiomatic Python prefera `<this> if <that> else <other>` rather than `<other> if not <that> else <this>` ...if you see what I mean.

Also ideally we would also check that 2 auth servers had been passed in the array....

```suggestion
        self.primary_auth = auths[0] if auths else None
        self.secondary_auth = auths[1] if auths else None

```